apps/web-app/src/context/SemaphoreContext.tsx

```diff
@@ -46,12 +62,31 @@ export default function useSemaphore(): SemaphoreContextType {
         [_feedback]
     )
 
-    return {
-        _users,
-        _feedback,
-        refreshUsers,
-        addUser,
-        refreshFeedback,
-        addFeedback
+    useEffect(() => {
+        refreshUsers()
+        refreshFeedback()
+    }, [refreshFeedback, refreshUsers])
+
+    return (
+        <SemaphoreContext.Provider
+            value={{
+                _users,
+                _feedback,
+                refreshUsers,
+                addUser,
+                refreshFeedback,
+                addFeedback
+            }}
+        >
+            {children}
+        </SemaphoreContext.Provider>
+    )
+}
+
+export const useSemaphoreContext = () => {
+    const context = useContext(SemaphoreContext)
+    if (context === null) {
```

Comment:
I think using "context === null" is better both for readability and type-checking because "!context" works if the context is 0, "" (empty string), null, undefined, false, or NaN. So, if we're explicitly checking for a null value, I believe the "context === null" approach is preferable. Considering this, let me know if you think using "!context" would be better, so I can make that change.

Reply:
Hey! I think that should work fine, but It would be nice to check if it's undefined too. You could also use `if (context == null)` to just check for null or undefined values.

Reply:
That's exactly why I think `context === null` is better; if someone changes the type to `undefined`, they'll realize that this if statement relies on a `null` check when they encounter compile errors, ensuring that typings remain consistent in the project.

Reply:
Hey! Why do you think that `context === null` (check just null) is better than `context == null` (check null and undefined) to throw the error?

Reply:
@alimahdiyar you're right. We only have to check if it's null there 👍